### PR TITLE
Add Julia parser generator

### PR DESCRIPTION
**Pull request by LeSeulArtichaut**
Companion PR to prologin/camisole#58.
Most code is "inspired", i.e. copy-pasted from the Python and Rust parsers :innocent:

### CODE REVIEW

File: iorgen/parser_julia.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+def gen_julia(input_data: Input, reprint: bool = False) -> str:
+    """Generate a Julia code to parse input"""
+    parser = ParserJulia(input_data)
+    for var in input_data.input:
+        parser.main.extend(parser.read_var(var))
+    parser.call(reprint)
+    return parser.content()
+
+
+RESERVED = [
```
Comment (Shaac) [inline iorgen/parser_julia.py:264]: You’ll probably need `parse` here as well (or in a second array if you want to keep the keywords seperated)

**Review by Shaac — COMMENTED** [review]
Very nice!
Thanks for the PR.
I’m surprised that the CI is all green since julia interpreter is not installed on the environment.
This shows issues on the test system!

I’ll review the PR more in depth over the weekend but it looks very good!

**Comment by LeSeulArtichaut** [comment]
> I’m surprised that the CI is all green since julia interpreter is not installed on the environment.
> This shows issues on the test system!

I've tested this and it looks like Julia is actually installed, probably as a dependency of some other package. I filed #32 to run CI with the `--no_missing` flag to actually enforce that all tested languages are installed. I'll add Julia as an explicit dependency as well.

**Comment by Shaac** [comment]
Running against the hidden extensive test suite (i.e. the prologin repository containing all the problems), I’ve encountered a puzzling error.

Minimal reproduction is:

```yaml
function_name: foo
input:
  - name: N
    type: int
    comment: $fail$
output: ""
```

And then running `echo 0 > foo.in && iorgen input.yaml -l jl --run foo.in`

It then fails with `ERROR: LoadError: syntax: invalid interpolation syntax: "$\n"`

It seems julia wants to parse things when it sees a $ in the comments. This is problematic and makes me wonder if it does other hidden stuff in the comments. So I guess `$` has to be escaped, but where exactly, in all the comments?

**Comment by LeSeulArtichaut** [comment]
That was because doc comments in Julia use regular strings (like in Python), and `$` is used for string interpolation. This should be fixed now.

**Comment by Shaac** [comment]
It is merged. Thanks!

**Pull request merged**